appyters/dimensionality_reduction_visualization/dimensionality_reduction_visualization.ipynb

```diff
@@ -0,0 +1,537 @@
+    "    img='load.png'\n",
+    "    \n",
+    ") %}\n",
+    "\n",
+    "\n",
+    "{% do SectionField(\n",
+    "    name='Visualization_Section',\n",
+    "    title='Select Visualization Parameters',\n",
+    "    subtitle='',\n",
+    "    img='select.png'\n",
+    "    \n",
+    ") %}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%appyter code_exec\n",
+    "{% set meta_data_filename = FileField(\n",
+    "    name='meta_data_filename', \n",
+    "    label='Meta data file (.csv or .txt)', \n",
+    "    default='mouse_brain_cells_metadata.csv',\n",
+    "\n",
+    "    examples={'mouse_brain_cells_metadata.csv': \"./GSE70466-metadata.txt\"}, \n",
+    "    description='Upload metadata as two-column comma seperated or tab seperated format. One column contains sample ids and the other column contains sample labels', \n",
+    "    section='Data_Section')\n",
+    "\n",
+    "%}\n",
+    "{% set rnaseq_data_filename = FileField(\n",
+    "    name='rnaseq_data_filename', \n",
+    "    label='RNA-seq data file (.csv or .txt)', \n",
+    "    default='mouse_brain_cells_gene_counts.csv',\n",
+    "    examples={'mouse_brain_cells_gene_counts.csv': \"./mouse_brain_cells_gene_counts\"}, \n",
+    "    description='Upload RNA-seq expression data as comma seperated or tab seperated format. The index of the dataset are genes, the columns are samples.', \n",
+    "    section='Data_Section')\n",
+    "\n",
+    "%}\n",
+    "{% set meta_class_column_name = StringField(\n",
+    "    name='meta_class_column_name', \n",
+    "    label='Class column name in metadata', \n",
+    "    default='cell_ontology_class', \n",
+    "    description='class column name of metadata', \n",
+    "    section='Data_Section')\n",
+    "%}\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%appyter code_exec\n",
+    "{% set plot_3D = ChoiceField(\n",
+    "    name='3D_plot', \n",
+    "    label='3D plots?', \n",
+    "    choices = {'2D': '2D', '3D and 2D': '3D'},\n",
+    "    default='3D and 2D', \n",
+    "    description='Check if User wants 3D plots', \n",
+    "    section='Visualization_Section')\n",
+    "%}\n",
+    "\n",
+    "{% set visualization_method = ChoiceField(\n",
+    "    name='visualization_method', \n",
+    "    label='Visualization Methods', \n",
+    "    choices = {'All': 'All','PCA': 'PCA', 'UMAP': 'UMAP', 't-SNE': 't-SNE'},\n",
+    "    default='All', \n",
+    "    description='Select a visualization method', \n",
+    "    section='Visualization_Section')\n",
+    "%}\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%appyter code_exec\n",
+    "rnaseq_data_filename = \"{{rnaseq_data_filename.value}}\"\n",
```

Comment:
Please use `{{rnaseq_data_filename}}`, quotes will be added automatically in a secure manner, same thing with the other fields.